# Guided Project: Profitable App Profiles for the App Store and Google Play Markets


The art of app profiling in an imaginary setting...
...but with a concrete result.

## project details

In the code cell below, we:

- Open the `AppleStore.csv` file using the `open()` function, and assign the output to a variable named `opened_file`
- Import the `reader()` function from the `csv` module
- Read in the opened file using the `reader()` function, and assign the output to a variable named `read_file`
- Transform the read-in file to a list of lists using `list()` and save it to a variable named `apps_data`
- Display the header row and the first three rows of the data set.

The data set above contains information about more than 7000 Apple iOS mobile apps. The data was collected from the iTunes Search API by data engineer [Ramanathan Perumal](https://www.kaggle.com/ramamet4). Documentation for the data set can be found [at this page](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home), where you'll also be able to download the data set.

This is a table explaining what each column in the data set describes:

Column name | Description
-- | --
"id" | App ID
"track_name"| App Name
"size_bytes"| Size (in Bytes)
"currency"| Currency Type
"price"| Price amount
"rating_count_tot"| User Rating counts (for all version)
"rating_count_ver"| User Rating counts (for current version)
"user_rating" | Average User Rating value (for all version)
"user_rating_ver"| Average User Rating value (for current version)
"ver" | Latest version code
"cont_rating"| Content Rating
"prime_genre"| Primary Genre
"sup_devices.num"| Number of supporting devices
"ipadSc_urls.num"| Number of screenshots showed for display
"lang.num"| Number of supported languages
"vpp_lic"| Vpp Device Based Licensing Enabled

In [98]:
from csv import reader

### The Google Play data set ###
# TO prevent UnicodeDecodeError, add encoding="utf8" inside open
opened_file = open('googleplaystore.csv', encoding="utf8") # T
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

In [99]:
### The App Store data set ###
opened_file = open('AppleStore.csv', encoding="utf8")
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [100]:
def explore_data(dataset, start =0, end = 2, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [101]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [102]:
explore_data(android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


- google play has around 10841 apps and columns
- most important columns include `'App'`, `'Category'`, `'Reviews'`, `'Installs'`, `'Type'`, `'Price'`, and `'Genres'`.

In [103]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [104]:
explore_data(ios, 0,4, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7197
Number of columns: 16


- we have 7197 apps in the ios data set
- some of the interesting columns include 'track_name', 'currency', 'price', 'rating_count_tot', 'user_rating', 'prime_genre'

- Details about each column can be found in the data set [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home)

## Deleting Wrong Data
- The Google Play data set has a dedicated [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion)
- one of the discussions outlines an error at row 10472
- let's find this error and compare it with the header


In [105]:
print(android[10472]) # incorrect row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [106]:
print(android_header) # header

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [107]:
print(android[0]) # correct row

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


- the row corresponds to app _Life Made WI-Fi Touchscreen Photo Frame_ where the rating is 19
- it can't be possible as the max rating is 5
- this error is caused by a missing value in the `'category'` column


In [108]:
for row in android[1:]:
    if len(row) != len(android_header):
        print(row)
        print("\n")
        print("Index postion is:", android.index(row))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


Index postion is: 10472


In [109]:
print(len(android))
del android[10472]  # run this just once
print(len(android))

10841
10840


## Removing Duplicate Entries

### Part One
we can observe that in Google Play data some apps have more than one entry.
For instance, the application Instagram has four entries:

In [110]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [111]:
duplicate_apps = []
unique_apps = []
for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print("Examples of duplicate apps: ", duplicate_apps[:15])  # Print first 15 duplicates</span>


Number of duplicate apps: 1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app.

We won't remove rows randomly, but rather we'll keep the rows that have the highest number of reviews because the higher the number of reviews, the more reliable the ratings.

To do that, we will:

- Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app
- Use the dictionary to create a new data set, which will have only one entry per app (and we only select the apps with the highest number of reviews)






### Part Two

Let's start by building the dictionary.



If you examine the rows we printed two cells above for the Instagram app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show that the data was collected at different times. We can use this to build a criterion for keeping rows.




In [112]:
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
    


we found that there are 1,181 cases where an app occurs more than once
- so the length of our dictionary (of unique apps) should be 
- equal to the difference between the length of our data set and 1,181.

In [113]:
print('Expected length: ', len(android) - 1181)
print('actual length: ', len(reviews_max))

Expected length:  9659
actual length:  9659


We start by initializing two empty lists, `android_clean` and `already_added`.
- We loop through the `android` data set, and for every iteration:
    - We isolate the name of the app and the number of reviews.
    - We add the current row (`app`) to the `android_clean` list, and the app name (`name`) to the `already_added` list if:
        - The number of reviews of the current app matches the number of reviews of that app as described in the `reviews_max` dictionary; and
        - The name of the app is not already in the `already_added` list. We need to add this supplementary condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry (for example, the Box app has three entries, and the number of reviews is the same). If we just check for `reviews_max[name] == n_reviews`, we'll still end up with duplicate entries for some apps.

In [114]:

android_clean = []
already_added = []
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if reviews_max[name] == n_reviews and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)




In [115]:
already_added[:1]

['Photo Editor & Candy Camera & Grid & ScrapBook']

In [116]:
android_clean[:1]

[['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up']]

In [117]:
len(android_clean)

9659

In [118]:
explore_data(android_clean, 0, 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9659
Number of columns: 13


We have 9659 rows, just as expected.

## Removing Non-English Apps

### Part One

If you explore the data sets enough, you'll notice the names of some of the apps suggest they are not directed toward an English-speaking audience.

In [119]:
print(ios[813][1])
print(ios[6731][1])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


### Part Two

To minimize the impact of data loss, we'll only remove an app if its name has more than three non-ASCII characters:

In [120]:
def is_english(word):
    """Check if a word is in English"""
    non_ascii = 0
    for char in word:
        if ord(char) > 127:
            non_ascii += 1
    if  non_ascii > 3:
        return False
    else:
        return True    


In [121]:
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


In [122]:
print(ord('™'))
print(ord('😜'))

8482
128540


In [123]:
print(is_english('Docs To Go™ Free Office Suite'))

True


In [124]:
android_english = []
ios_english = []
for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
    
for app in ios:
    name = app[1]    
    if is_english(name):
        ios_english.append(app)
    

In [125]:
explore_data(android_english, 0,1,True)
print('\n')
explore_data(ios_english, 0,1, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


Number of rows: 6183
Number of columns: 16


## Isolating the Free Apps
Our data sets contain both free and non-free apps, and we'll need to isolate only the free apps for our analysis. Below, we isolate the free apps for both our data sets.


In [126]:
android_final = []
ios_final = []
for app in android_english:
   price  = app[7]
   if price =='0':
       android_final.append(app)
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)


In [127]:
print(len(android_final))
print(len(ios_final))

8864
3222


We're left with 8864 Android apps and 3222 iOS apps, which should be enough for our analysis.

## Most Common Apps by Genre
our goal is to determine the kinds of apps that are likely to attract more users because the number of people using our apps affect our revenue.

To minimize risks and overhead, our validation strategy for an app idea has three steps:

Build a minimal Android version of the app, and add it to Google Play.
If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

we'll build a frequency table for the `prime_genre` column of the App Store data set, and the `Genres` and `Category` columns of the Google Play data set.

We'll build two functions we can use to analyze the frequency tables:

- One function to generate frequency tables that show percentages
- Another function that we can use to display the percentages in a descending order

In [137]:
def freq_table(dataset, index):
    table ={}
    total = 0
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1 
    table_percentage = {}
    for key in table:
        percentage = (table[key] / total ) * 100
        table_percentage[key] = percentage
        
    return table_percentage   
        
def display_table(dataset, index):
        table = freq_table(dataset, index)
        table_display = []
        for key in table:
            key_value_as_tuple = (table[key], key)
            table_display.append(key_value_as_tuple)
        
        table_sorted = sorted(table_display, reverse=True)
        for item in table_sorted:
            print(item[1], ": ", item[0])
            
    

In [140]:
display_table(android_final, 1) # Category

FAMILY :  18.907942238267147
GAME :  9.724729241877256
TOOLS :  8.461191335740072
BUSINESS :  4.591606498194946
LIFESTYLE :  3.9034296028880866
PRODUCTIVITY :  3.892148014440433
FINANCE :  3.7003610108303246
MEDICAL :  3.531137184115524
SPORTS :  3.395758122743682
PERSONALIZATION :  3.3167870036101084
COMMUNICATION :  3.2378158844765346
HEALTH_AND_FITNESS :  3.0798736462093865
PHOTOGRAPHY :  2.944494584837545
NEWS_AND_MAGAZINES :  2.7978339350180503
SOCIAL :  2.6624548736462095
TRAVEL_AND_LOCAL :  2.33528880866426
SHOPPING :  2.2450361010830324
BOOKS_AND_REFERENCE :  2.1435018050541514
DATING :  1.861462093862816
VIDEO_PLAYERS :  1.7937725631768955
MAPS_AND_NAVIGATION :  1.3989169675090252
FOOD_AND_DRINK :  1.2409747292418771
EDUCATION :  1.1620036101083033
ENTERTAINMENT :  0.9589350180505415
LIBRARIES_AND_DEMO :  0.9363718411552346
AUTO_AND_VEHICLES :  0.9250902527075812
HOUSE_AND_HOME :  0.8235559566787004
WEATHER :  0.8009927797833934
EVENTS :  0.7107400722021661
PARENTING :  0.6543

In [139]:
display_table(ios_final, -5)

Games :  58.16263190564867
Entertainment :  7.883302296710118
Photo & Video :  4.9658597144630665
Education :  3.662321539416512
Social Networking :  3.2898820608317814
Shopping :  2.60707635009311
Utilities :  2.5139664804469275
Sports :  2.1415270018621975
Music :  2.0484171322160147
Health & Fitness :  2.0173805090006205
Productivity :  1.7380509000620732
Lifestyle :  1.5828677839851024
News :  1.3345747982619491
Travel :  1.2414649286157666
Finance :  1.1173184357541899
Weather :  0.8690254500310366
Food & Drink :  0.8069522036002483
Reference :  0.5586592178770949
Business :  0.5276225946617008
Book :  0.4345127250155183
Navigation :  0.186219739292365
Medical :  0.186219739292365
Catalogs :  0.12414649286157665


In [141]:
display_table(android_final, -4)

Tools :  8.449909747292418
Entertainment :  6.069494584837545
Education :  5.347472924187725
Business :  4.591606498194946
Productivity :  3.892148014440433
Lifestyle :  3.892148014440433
Finance :  3.7003610108303246
Medical :  3.531137184115524
Sports :  3.463447653429603
Personalization :  3.3167870036101084
Communication :  3.2378158844765346
Action :  3.1024368231046933
Health & Fitness :  3.0798736462093865
Photography :  2.944494584837545
News & Magazines :  2.7978339350180503
Social :  2.6624548736462095
Travel & Local :  2.3240072202166067
Shopping :  2.2450361010830324
Books & Reference :  2.1435018050541514
Simulation :  2.0419675090252705
Dating :  1.861462093862816
Arcade :  1.8501805054151623
Video Players & Editors :  1.7712093862815883
Casual :  1.7599277978339352
Maps & Navigation :  1.3989169675090252
Food & Drink :  1.2409747292418771
Puzzle :  1.128158844765343
Racing :  0.9927797833935018
Role Playing :  0.9363718411552346
Libraries & Demo :  0.9363718411552346
Aut

- we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps.
- The difference between the `Genres` and the `Category` columns is not crystal clear
- In IOS, the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%.
- The frequency tables showed us that apps designed for fun dominate the App Store, while Google Play shows a more balanced landscape of both practical and fun apps.

## Most Popular Apps by Genre on the App Store
- One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre
- For the Google Play data set, we can find this information in the `Installs` column
- the App Store data set this information is missing
- we can use the total number of user ratings as a proxy, which we can find in the `rating_count_tot` app

In [151]:
genre_ios = freq_table(ios_final, -5)
genre_ios

{'Social Networking': 3.2898820608317814,
 'Photo & Video': 4.9658597144630665,
 'Games': 58.16263190564867,
 'Music': 2.0484171322160147,
 'Reference': 0.5586592178770949,
 'Health & Fitness': 2.0173805090006205,
 'Weather': 0.8690254500310366,
 'Utilities': 2.5139664804469275,
 'Travel': 1.2414649286157666,
 'Shopping': 2.60707635009311,
 'News': 1.3345747982619491,
 'Navigation': 0.186219739292365,
 'Lifestyle': 1.5828677839851024,
 'Entertainment': 7.883302296710118,
 'Food & Drink': 0.8069522036002483,
 'Sports': 2.1415270018621975,
 'Book': 0.4345127250155183,
 'Finance': 1.1173184357541899,
 'Education': 3.662321539416512,
 'Productivity': 1.7380509000620732,
 'Business': 0.5276225946617008,
 'Catalogs': 0.12414649286157665,
 'Medical': 0.186219739292365}

In [152]:
ios_final[1][5]

'2161558'

In [153]:
ios_final[1]

['389801252',
 'Instagram',
 '113954816',
 'USD',
 '0.0',
 '2161558',
 '1289',
 '4.5',
 '4.0',
 '10.23',
 '12+',
 'Photo & Video',
 '37',
 '0',
 '29',
 '1']

- we calculate the average number of user ratings per app genre on the App Store:

In [156]:
for genre in genre_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(f'{genre} :  {avg_n_ratings}')
            

Social Networking :  71548.34905660378
Photo & Video :  28441.54375
Games :  22788.6696905016
Music :  57326.530303030304
Reference :  74942.11111111111
Health & Fitness :  23298.015384615384
Weather :  52279.892857142855
Utilities :  18684.456790123455
Travel :  28243.8
Shopping :  26919.690476190477
News :  21248.023255813954
Navigation :  86090.33333333333
Lifestyle :  16485.764705882353
Entertainment :  14029.830708661417
Food & Drink :  33333.92307692308
Sports :  23008.898550724636
Book :  39758.5
Finance :  31467.944444444445
Education :  7003.983050847458
Productivity :  21028.410714285714
Business :  7491.117647058823
Catalogs :  4004.0
Medical :  612.0


On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [157]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ': ', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic :  345046
Google Maps - Navigation & Transit :  154911
Geocaching® :  12811
CoPilot GPS – Car Navigation & Offline Maps :  3582
ImmobilienScout24: Real Estate Search in Germany :  187
Railway Route Search :  5


The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [158]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

- Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

- Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

- Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

## Most Popular Apps by Genre on Google Play

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [159]:
display_table(android_final, 5) # the Installs columns

1,000,000+ :  15.726534296028879
100,000+ :  11.552346570397113
10,000,000+ :  10.548285198555957
10,000+ :  10.198555956678701
1,000+ :  8.393501805054152
100+ :  6.915613718411552
5,000,000+ :  6.825361010830325
500,000+ :  5.561823104693141
50,000+ :  4.7721119133574
5,000+ :  4.512635379061372
10+ :  3.5424187725631766
500+ :  3.2490974729241873
50,000,000+ :  2.3014440433213
100,000,000+ :  2.1322202166064983
50+ :  1.917870036101083
5+ :  0.78971119133574
1+ :  0.5076714801444043
500,000,000+ :  0.2707581227436823
1,000,000,000+ :  0.22563176895306858
0+ :  0.04512635379061372
0 :  0.01128158844765343
